compiler/dyno/tools/chpldoc/chpldoc.cpp

```diff
@@ -690,190 +853,59 @@ struct RstSignatureVisitor {
-    bool addSpace = wantSpaces(node->op(), printingType_) ||
-                    node->op() == USTR("by") ||
-                    node->op() == USTR("align") ||
-                    node->op() == USTR("reduce=") ||
-                    node->op() == USTR("reduce") ||
-                    node->op() == USTR("scan") ||
-                    node->op() == USTR("dmapped");
-    opHelper(node, 0, false);
-    if (addSpace && node->op() != USTR(":"))
-      os_ << " ";
-    os_ << node->op();
-    if (addSpace)
-      os_ << " ";
-    opHelper(node, 1, false);
-  }
-
-  bool enter(const OpCall* node) {
-    if (node->isUnaryOp()) {
-      printUnaryOp(node);
-    } else if (node->isBinaryOp()) {
-      printBinaryOp(node);
-    }
-    return false;
-  }
-
-  bool enter(const FnCall* call) {
-    const AstNode* callee = call->calledExpression();
-    assert(callee);
-    callee->traverse(*this);
-    if (isCalleReservedWord(callee)) {
-      os_ << " ";
-      call->actual(0)->traverse(*this);
-    } else {
-      if (call->callUsedSquareBrackets()) {
-        os_ << "[";
-      } else {
-        os_ << "(";
-      }
-      std::string sep;
-      for (int i = 0; i < call->numActuals(); i++) {
-        os_ << sep;
-        if (call->isNamedActual(i)) {
-          os_ << call->actualName(i);
-          // The spaces around = are just to satisfy old tests
-          // TODO: Remove spaces around `=` when removing old parser
-          os_ << " = ";
-        }
+    // Return type
+    if (const AstNode* e = f->returnType()) {
+      os_ << ": ";
+      printingType_ = true;
+      e->traverse(*this);
+      printingType_ = false;
+    }
 
-        call->actual(i)->traverse(*this);
+    // throws
+    if (f->throws()) os_ << " throws";
 
-        sep = ", ";
-      }
-      if (call->callUsedSquareBrackets()) {
-        os_ << "]";
-      } else {
-        os_ << ")";
-      }
-    }
     return false;
   }
 
-  bool enter(const Domain* dom) {
-    interpose(dom->children(), ", ", "{", "}");
+  bool enter(const Identifier* r) {
+    os_ << r->name().c_str();
     return false;
   }
 
-  bool enter(const Tuple* tup) {
-    os_ << "(";
-    bool first = true;
-    for (auto it = tup->children().begin(); it != tup->children().end(); it++) {
-      if (!first) os_ << ", ";
-      (*it)->traverse(*this);
-      first = false;
-    }
-    if (tup->numChildren() == 1) {
-      os_ << ",";
+  bool enter(const Import* node) {
+    node->stringify(os_, StringifyKind::CHPL_SYNTAX);
+    return false;
+  }
+
+  bool enter(const New* d) {
+    os_ << "new ";
+    if (d->management() != New::Management::DEFAULT_MANAGEMENT) {
+      os_ << kindToString(d->management()) << " ";
     }
-    os_ << ")";
+    d->typeExpression()->traverse(*this);
     return false;
   }
 
-  bool enter(const Array* arr) {
-    interpose(arr->children(), ", ", "[", "]");
+  /* Numeric Literals */
+  // TODO: Truncate precision of floating point literals to
+  //  match chpldoc's precision.
```

Comment:
We don't want to do that, so update this comment in the follow-up PR.

Reply:
It's updated in the next PR